R/applyFilt.R

```diff
@@ -2024,55 +2024,125 @@ pmartR_filter_worker <- function (filter_object, omicsData) {
     # remove any samples from f_data and e_data #
     if (!is.null(filter_object$f_data_remove)) {
 
+      valid_names <- which(names(omicsData$f_data) == samp_cname)
+      
+      # Warn if trying to remove samples which are no longer in the data
+      # missing_samples <- unlist(
+      #   lapply(filter_object$f_data_remove, function(x) if (!(x %in% valid_names)) x)
+      # )
+      # 
+      # if (length(missing_samples > 0)) {
+      #   warning("Specified samples ",
+      #           paste0(missing_samples, collapse = ", "),
+      #           " were not found in the f_data.")
+      # }
+      
       # Find the row indices of the sample names that will be removed from the
       # f_data object.
       inds <- which(
-        omicsData$f_data[, which(names(omicsData$f_data) == samp_cname)]
+        omicsData$f_data[, valid_names]
         %in% filter_object$f_data_remove
       )
 
-      # Remove the rows in f_data that correspond to the samples that should be
-      # removed.
-      omicsData$f_data <- omicsData$f_data[-inds, ]
+      # Check if there is at least one sample name that will be removed
+      if (length(inds) > 0) {
+        
+        # Remove the rows in f_data that correspond to the samples that should
+        # be removed.
+        omicsData$f_data <- omicsData$f_data[-inds, ]
+        
+      }
 
+      valid_names <- names(omicsData$e_data)
```

Comment:
leave out edata_cname here

Reply:
What do you mean by this?

Reply:
Is this what you mean? 
```R
valid_names <- names(omicsData$e_data)
valid_names <- valid_names[valid_names != edata_cname]
```

Reply:
If I'm reading this correctly `valid_names` will include the identifier column specified by `attr(object, "cnames")$edata_cname` (usually the first column), but only the names of the other columns are actual sample names to be potentially removed.  Probably doesn't matter, but I think this should be:

`valid_names <- names(omicsData$e_data[-which(names(omicsData$e_data) == edata_cname)])`

or

`valid_names <- omicsData$f_data[,get_fdata_cname(omicsData)]`

Reply:
Hmm... it seems like changing it is causing a lot more test failures

Reply:
Yea I'm seeing these, looking into it

Reply:
Sorry yea this borks the indexing.  I think it's fine actually, just weird because the first element of valid_names isn't actually a valid name, but is needed for the right indexing.